File: unit2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
    char *_cipher = argv[1];
    int cipher = atoi(_cipher);
    char *input = get_string("plaintext: ");
    int stringLength = strlen(input);
    int store;
    int modified;
    bool capital;
    bool space = false;
    bool done;
    printf("ciphertext: ");
    for (int i = 0; i < stringLength; i++)
    {
        done = false;
        if (65 <= input[i] && input[i] <= 90)
        {
            capital = true;
        }
        else if (97 <= input[i] && input[i] <= 122)
        {
            capital = false;
        }
        else
        {
            goto no;
        }
        modified = input[i] + cipher;
        while (modified > 90 && capital == true && space == false)
        {
            modified -= 26;
        }
        while (modified > 122 && capital == false && space == false)
        {
            modified -= 26;
        }
        printf("%c", modified);
        done = true;
no:
        if (!done)
        {
            printf("%c", input[i]);
        }
    }
    printf("\n");
    return 0;
}

File: unit6/sentimental/caesar.py
from cs50 import get_string
from string import ascii_lowercase as lc
from string import ascii_uppercase as uc
import sys

if (len(sys.argv) != 2):
    sys.exit(1)

cipher = int(sys.argv[1])
s = get_string("plaintext: ")
ns = ''
i = 0
for c in s:
    if c in lc:
        ns = ns + lc[(lc.index(c)+cipher) % len(lc)]
    elif c in uc:
        ns = ns + uc[(uc.index(c)+cipher) % len(uc)]
    else:
        ns += c
print("ciphertext: " + ns)